extutils/imgproc/apng2png.py

```diff
@@ -0,0 +1,285 @@
+
+        return _DataChunkIHDR(_BaseChunk.make_chunk("IHDR", new_data))
+
+
+@dataclass
+class _DataChunkacTL(_DataChunkBase):
+    frame_count: int = field(init=False)
+
+    def __post_init__(self):
+        super().__post_init__()
+
+        self.frame_count = struct.unpack(">I", self.pure_data[0:4])[0]
+
+
+@dataclass
+class _DataChunkIDAT(_DataChunkBase):
+    pass
+
+
+@dataclass
+class _DataChunkIEND(_DataChunkBase):
+    pass
+
+
+@dataclass
+class _DataChunkOther(_DataChunkBase):
+    pass
+
+
+@dataclass
+class _DataChunkfcTL(_DataChunkBase):
+    seq_num: int = field(init=False)
+    width: int = field(init=False)
+    height: int = field(init=False)
+    x_offset: int = field(init=False)
+    y_offset: int = field(init=False)
+    delay_num: int = field(init=False)
+    delay_den: int = field(init=False)
+    dispose: _Dispose = field(init=False)
+    blend: int = field(init=False)
+
+    def __post_init__(self):
+        super().__post_init__()
+
+        self.seq_num, self.width, self.height, self.x_offset, self.y_offset = \
+            struct.unpack(">IIIII", self.pure_data[0:20])
+        self.delay_num, self.delay_den, self.dispose, self.blend = \
+            struct.unpack(">HHBB", self.pure_data[20:26])
+
+        self.dispose = _Dispose(self.dispose)
+
+    # noinspection PyPep8Naming
+    def generate_IHDR(self, IHDR: _DataChunkIHDR) -> _DataChunkIHDR:
+        return IHDR.replace_resolution(self.width, self.height)
+
+
+@dataclass
+class _DataChunkfdAT(_DataChunkBase):
+    # noinspection PyPep8Naming
+    def to_IDAT(self):
+        # Skipping 4 B because it's frame number
+        return _DataChunkIDAT(_BaseChunk.make_chunk("IDAT", self.base_chunk.pure_data[4:]))
+
+
+def _parse_chunks(data) -> Generator[_BaseChunk, None, None]:
+    data = data[8:]
+
+    while data:
+        new_chunk = _BaseChunk.from_bytes(data)
+        yield new_chunk
+
+        data = data[new_chunk.length + 12:]
+
+
+# noinspection PyPep8Naming
+def _create_image_bytes(IHDR: _DataChunkIHDR, fcTL: _DataChunkfcTL,
+                        fdAT_IDAT: Union[_DataChunkfdAT, _DataChunkIDAT], IEND: _DataChunkIEND):
+    # Append signature
+    image_data = png_signature
+
+    # Append new header by size info from fcTC
+    image_data += fcTL.generate_IHDR(IHDR).data
+
+    if hasattr(fdAT_IDAT, "to_IDAT"):
+        # fdAT
+        image_data += fdAT_IDAT.to_IDAT().data
+    else:
+        # IDAT
+        image_data += fdAT_IDAT.data
+
+    # Append IEND
+    image_data += IEND.data
+
+    return image_data
+
+
+def extract_frames(apng_path: str) -> List[Tuple[bytes, Fraction]]:
+    """
+    Extract the frames of apng file at ``apng_path``.
+
+    Returns a 2-tuple containing the byte data of a frame and its delay.
+
+    :param apng_path: path of the apng file to be extracted
+    :returns: list of 2-tuple containing frame byte data and its delay
+    """
+    with open(apng_path, "rb") as f:
```

Comment:
Codacy found an issue: [Variable name "f" doesn't conform to snake_case naming style](https://app.codacy.com/manual/RaenonX/Jelly-Bot/pullRequest?prid=6015490)